src/psyclone/dynamo0p3.py

```diff
@@ -7348,11 +7348,22 @@ def independent_iterations(self,
 
         # The generic DA says that this loop cannot be parallelised. However,
         # we use domain-specific information to qualify this.
-        if self.loop_type in ["colour", "dof"]:
+        if self.loop_type == "colour":
             # This loop is either over cells of a single colour or DoFs.
             # According to LFRic rules this is safe to parallelise.
             return True
 
+        if self.loop_type == "dof":
+            # The generic DA can't see the PSyIR of this Builtin (because it
+            # hasn't been lowered to language level) so we use
+            # domain-specific knowledge about its properties.
+            if self.kernel.is_reduction:
+                dtools._add_message(
+                    f"Builtin '{self.kernel.name}' performs a reduction",
+                    DTCode.WARN_SCALAR_REDUCTION)
+                return False
+            return True
```

Comment:
Just to check, is this different from the condition below, does bulitins have "has_inc_arg"? and does the "has_inc_arg" below show put a similar message in dtools?

Reply:
No, Builtins do not have `has_inc_arg` as they do not perform INC operations (which only happen when we iterate over cells and write to shared dofs). However, they can perform reductions (e.g. MIN, MAX, SUM).
I've extended the "has_inc_arg" so it too adds a message to dtools. BTW, I'm currently being naughty and doing `dtools._add_message`. Should I make the `add_message` method public? Is @hiker happy with that?

Reply:
Gee, that's an odd case. To some degree I wouldn't want to make this public (since this functionality should not be used by someone using the DT), but it makes sense here :(
TBH, I don't mind either. The `independent_iterations` function is kind of part of the DT, so I would be ok leaving it this way (i.e. 'private'), but I would think that you also need to disable a warning here(?).The moment a user-script would add its own messages (e.g. in case of a slightly-higher-level script that's being used by a PSyclone transformation), this should then become public.